api/server/service.go

```diff
@@ -10,25 +10,21 @@ import (
 )
 
 type Service struct {
-	rti     runtime.Initializer
-	aid     string // account ID
-	cid     string // caller ID
-	builder builder.Builder
-	vault   vault.Vault
+	rti   runtime.Initializer
+	aid   string // account ID
+	cid   string // caller ID
+	vault vault.Vault
 
 	Builder *BuilderService
 }
 
 func (s *Service) InitializeBuilder(ctx context.Context, builder string) (builder.Builder, error) {
-	if s.builder != nil {
-		return s.builder, nil
-	}
 	bld, err := s.rti.InitializeBuilder(ctx, s.cid, builder)
```

Comment:
The runtime.Initializer is really old code. The newer pattern is the [service pattern](https://github.com/unweave/unweave/tree/master/services) used by `execs`, `volumes` etc.  We should probably dot he same for the builder. i.e. create a `buildersrv` and inject that into the `main.go`

Reply:
@noorvir can we tackle it in a follow-up PR?

That refactor might be quite sizeable, and is unrelated to the goal of this PR.